packages/front/src/lib/resources/associations/association.port.ts

```diff
@@ -1,25 +1,29 @@
-import type { Association, DocumentDto, PaginatedAssociationNameDto } from "dto";
+import type { Association, DocumentDto, Grant, PaginatedAssociationNameDto, AssociationIdentifiers } from "dto";
 import requestsService from "$lib/services/requests.service";
 
 class AssociationPort {
+    getResource(identifier: AssociationIdentifiers, resource?: string) {
+        return requestsService.get(`/association/${identifier}${resource ? "/" + resource : ""}`);
+    }
+
     incExtractData(identifier) {
-        const path = `/association/${identifier}/extract-data`;
-        requestsService.get(path).catch(() => null);
+        this.getResource(identifier, "extract-data").catch(() => null);
+    }
+
+    async getByIdentifier(identifier: AssociationIdentifiers): Promise<Association | undefined> {
+        return (await this.getResource(identifier))?.data?.association;
     }
 
-    async getByIdentifier(identifier): Promise<Association | undefined> {
-        const path = `/association/${identifier}`;
-        return (await requestsService.get(path))?.data?.association;
+    async getEstablishments(identifier: AssociationIdentifiers) {
+        return (await this.getResource(identifier, "etablissements"))?.data?.etablissements;
     }
 
-    async getEstablishments(identifier) {
-        const path = `/association/${identifier}/etablissements`;
-        return (await requestsService.get(path))?.data?.etablissements;
+    async getGrants(identifier: AssociationIdentifiers): Promise<Grant[]> {
+        return (await this.getResource(identifier, "grants"))?.data?.subventions;
     }
 
-    async getDocuments(identifier): Promise<DocumentDto[]> {
-        const path = `association/${identifier}/documents`;
-        return (await requestsService.get(path))?.data?.documents;
+    async getDocuments(identifier: AssociationIdentifiers): Promise<DocumentDto[]> {
+        return (await this.getResource(identifier, "documents"))?.data?.documents;
```

Comment:
```suggestion
        return (await this.getResource(identifier, "documents"))?.data?.documents || [];
```
sinon ya pas de valeur par défaut

Reply:
Je me rends compte que yaurait besoin de ça ailleurs

Reply:
j'ai répondu plus haut. En plus c'est hors scope et si l'API s'occupe déjà de définir un valeur par défaut alors pas besoin (cf utilisation des DTO de réponse)

Reply:
Mais je vois bien que j'ai pas compris ta réponse alors

Reply:
comme dis plus haut : si l'API renvoi pas un tableau, alors c'est qu'on doit soulever une erreur (et pas mettre par défaut ce qui cache l'erreur)